packages/api-explorer/src/components/SideNav/SideNav.tsx

```diff
@@ -111,15 +110,25 @@ export const SideNav: FC<SideNavProps> = ({ headless = false, spec }) => {
     setSearchPattern(value)
   }
 
+  useEffect(() => {
```

Comment:
Move 
`const searchParams = new URLSearchParams(location.search)`
here?
Means location.search is a dependency.

Reply:
This makes sense, but why is location.search a necessary dependency if we are using the `navigate` hook in the effect; wouldn't this cause repeated calls since location.search is prone to change with new query parameters? There is another effect in `ApiExplorer.tsx` with `location.search` as a dependency and makes changes accordingly.